apps/passport-client/components/screens/ChangePasswordScreen.tsx

```diff
@@ -0,0 +1,161 @@
+import { PCDCrypto } from "@pcd/passport-crypto";
+import { requestPasswordSalt } from "@pcd/passport-interface";
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+import { appConfig } from "../../src/appConfig";
+import { useDispatch, useSelf } from "../../src/appHooks";
+import { saveEncryptionKey } from "../../src/localstorage";
+import { updateStorage, uploadStorage } from "../../src/useSyncE2EEStorage";
+import { CenterColumn, H2, Spacer, TextCenter } from "../core";
+import { LinkButton } from "../core/Button";
+import { MaybeModal } from "../modals/Modal";
+import { AppContainer } from "../shared/AppContainer";
+import { NewPasswordForm, PasswordInput } from "../shared/NewPasswordForm";
+
+export function ChangePasswordScreen() {
+  const self = useSelf();
+  const navigate = useNavigate();
+  const [loading, setLoading] = useState(false);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (self == null) {
+      navigate("/login", { replace: true });
+    }
+  }, [self, navigate]);
+
+  const [currentPassword, setCurrentPassword] = useState("");
+  const [newPassword, setNewPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [revealPassword, setRevealPassword] = useState(false);
+
+  const onChangePassword = async () => {
+    if (loading) return;
+    try {
+      const saltResult = await requestPasswordSalt(
+        appConfig.passportServer,
+        self.email
+      );
+      if (!saltResult.success) {
+        throw new Error("Error occurred while fetching salt from server");
+      }
+
+      const crypto = await PCDCrypto.newInstance();
+      const currentEncryptionKey = await crypto.argon2(
+        currentPassword,
+        saltResult.value,
+        32
+      );
+      const newSalt = await crypto.generateSalt();
+      const newEncryptionKey = await crypto.argon2(newPassword, newSalt, 32);
+      const res = await updateStorage(
```

Comment:
`updateStorage` is a highly generic name that leaves me confused as to what is going on here